cmd/iceberg/main.go

```diff
@@ -60,52 +61,59 @@ Options:
   --credential TEXT  specify credentials for the catalog
   --warehouse TEXT   specify the warehouse to use`
 
+type Config struct {
+	List     bool `docopt:"list"`
+	Describe bool `docopt:"describe"`
+	Schema   bool `docopt:"schema"`
+	Spec     bool `docopt:"spec"`
+	Uuid     bool `docopt:"uuid"`
+	Location bool `docopt:"location"`
+	Props    bool `docopt:"properties"`
+	Drop     bool `docopt:"drop"`
+	Files    bool `docopt:"files"`
+	Rename   bool `docopt:"rename"`
+
+	Get    bool `docopt:"get"`
+	Set    bool `docopt:"set"`
+	Remove bool `docopt:"remove"`
+
+	Namespace bool `docopt:"namespace"`
+	Table     bool `docopt:"table"`
+
+	RenameFrom string `docopt:"<from>"`
+	RenameTo   string `docopt:"<to>"`
+
+	Parent   string `docopt:"PARENT"`
+	Ident    string `docopt:"IDENTIFIER"`
+	TableID  string `docopt:"TABLE_ID"`
+	PropName string `docopt:"PROPNAME"`
+	Value    string `docopt:"VALUE"`
+
+	Catalog   string `docopt:"--catalog"`
+	URI       string `docopt:"--uri"`
+	Output    string `docopt:"--output"`
+	History   bool   `docopt:"--history"`
+	Cred      string `docopt:"--credential"`
+	Warehouse string `docopt:"--warehouse"`
+}
+
 func main() {
 	args, err := docopt.ParseArgs(usage, os.Args[1:], iceberg.Version())
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	cfg := struct {
-		List     bool `docopt:"list"`
-		Describe bool `docopt:"describe"`
-		Schema   bool `docopt:"schema"`
-		Spec     bool `docopt:"spec"`
-		Uuid     bool `docopt:"uuid"`
-		Location bool `docopt:"location"`
-		Props    bool `docopt:"properties"`
-		Drop     bool `docopt:"drop"`
-		Files    bool `docopt:"files"`
-		Rename   bool `docopt:"rename"`
-
-		Get    bool `docopt:"get"`
-		Set    bool `docopt:"set"`
-		Remove bool `docopt:"remove"`
-
-		Namespace bool `docopt:"namespace"`
-		Table     bool `docopt:"table"`
-
-		RenameFrom string `docopt:"<from>"`
-		RenameTo   string `docopt:"<to>"`
-
-		Parent   string `docopt:"PARENT"`
-		Ident    string `docopt:"IDENTIFIER"`
-		TableID  string `docopt:"TABLE_ID"`
-		PropName string `docopt:"PROPNAME"`
-		Value    string `docopt:"VALUE"`
-
-		Catalog   string `docopt:"--catalog"`
-		URI       string `docopt:"--uri"`
-		Output    string `docopt:"--output"`
-		History   bool   `docopt:"--history"`
-		Cred      string `docopt:"--credential"`
-		Warehouse string `docopt:"--warehouse"`
-	}{}
+	cfg := Config{}
 
 	if err := args.Bind(&cfg); err != nil {
 		log.Fatal(err)
 	}
```

Comment:
should we probably add an option to specify the config file if they want something other than the default `.iceberg-go.yaml`?

Reply:
@zeroshade I have added `--config` option to specify the path to the configuration file.